File: samc-sort.java
import java.util.*;

public class SamcSort {
    static Random rand = new Random();

    public static List<Integer> sort(List<Integer> arr, int depth) {
        if (arr.size() <= 1) return arr;

        String indent = "  ".repeat(depth);
        System.out.println(indent + "<!-- Sorting " + arr + " -->");

        int pivot = Collections.max(arr);
        System.out.println(indent + "<h2 class='pivot'>Chosen Alpha: " + pivot + "</h2>");

        List<Integer> left = new ArrayList<>();
        List<Integer> right = new ArrayList<>();

        for (int num : arr) {
            if (num == pivot) continue;
            if (num < pivot) {
                System.out.println(indent + "<p class='insult'>" + randomInsult() + " (" + num + ")</p>");
                left.add(num);
            } else {
                System.out.println(indent + "<p class='retort'>" + randomRetort() + " (" + num + ")</p>");
                right.add(0, num);
            }
        }

        List<Integer> sorted = new ArrayList<>();
        sorted.addAll(sort(left, depth + 1));
        sorted.add(pivot);
        sorted.addAll(sort(right, depth + 1));
        return sorted;
    }

    static String randomInsult() {
        String[] insults = {
            "Pathetic. Go back to school.",
            "Your value is trash.",
            "I’ve seen better values in a Drupal theme."
        };
        return insults[rand.nextInt(insults.length)];
    }

    static String randomRetort() {
        String[] retorts = {
            "You think you're better than me?",
            "Nice try.",
            "Overcompensating, aren't you?"
        };
        return retorts[rand.nextInt(retorts.length)];
    }
}

File: samc-sort.rs
fn samc_sort(arr: Vec<i32>, depth: usize) -> Vec<i32> {
    let indent = "  ".repeat(depth);
    if arr.len() <= 1 {
        return arr;
    }

    println!("{}<!-- Sorting {:?} -->", indent, arr);
    let pivot = *arr.iter().max().unwrap();
    println!("{}<h2 class='pivot'>Chosen Alpha: {}</h2>", indent, pivot);

    let mut left = Vec::new();
    let mut right = Vec::new();

    for &num in arr.iter() {
        if num == pivot {
            continue;
        }
        if num < pivot {
            println!("{}<p class='insult'>{} ({})</p>", indent, random_insult(), num);
            left.push(num);
        } else {
            println!("{}<p class='retort'>{} ({})</p>", indent, random_retort(), num);
            right.insert(0, num);
        }
    }

    let mut sorted = samc_sort(left, depth + 1);
    sorted.push(pivot);
    sorted.extend(samc_sort(right, depth + 1));
    sorted
}

fn random_insult() -> &'static str {
    let insults = [
        "You disgust me, number.",
        "Not even close to Amazon.",
        "Your value is embarrassing."
    ];
    let i = rand::random::<usize>() % insults.len();
    insults[i]
}

fn random_retort() -> &'static str {
    let retorts = [
        "Overcompensating?",
        "Get back in your lane.",
        "You wish you were better."
    ];
    let i = rand::random::<usize>() % retorts.len();
    retorts[i]
}
